source/fundamentals/aggregation.txt

```diff
@@ -0,0 +1,237 @@
+method.
+
+The Java driver provides the `Aggregates <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html>`__
+helper class that
+contains builders for aggregation stages.
+
+In the following example, the aggregation pipeline:
+
+- Uses a :manual:`$match <reference/operator/aggregation/match/>` stage to filter for documents whose
+  ``categories`` array field contains the element ``Bakery``. The example uses
+  `Aggregates.match <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html#match(org.bson.conversions.Bson)>`__
+  to build the ``$match`` stage.
+
+- Uses  a :manual:`$group <reference/operator/aggregation/group/>` stage to group the matching documents by the ``stars``
+  field, accumulating a count of documents for each distinct value of ``stars``.
+
+  The example uses `Aggregates.group <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html#group(org.bson.conversions.Bson)>`__
+  to build the ``$group`` stage and `Accumulators.sum <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Accumulators#sum(java.lang.String,TExpression)>`__
+  to build the accumulator expression. The :manual:`accumulator expressions <reference/operator/aggregation/#accumulators-group>`
+  used within the ``$group`` stage are built with the `Accumulators <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Accumulators.html>`__
+  helper class.
+
+.. code-block:: java
+
+   collection.aggregate(
+         Arrays.asList(
+                 Aggregates.match(Filters.eq("categories", "Bakery")),
+                 Aggregates.group("$stars", Accumulators.sum("count", 1))
+         )
+   ).forEach(printBlock);
+
+.. code-block:: json
+
+   {"_id": 4, "count": 2}
+   {"_id": 5, "count": 1}
+
+Use Aggregation Expressions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Java driver provides builders for :java-sync-api:`accumulator expressions<com/mongodb/client/model/Accumulators.html>`
+for use with :java-sync-api:`$group <com/mongodb/client/model/Aggregates.html#group(TExpression,java.util.List)>`. You
+must declare all other expressions in JSON format or compatible document format.
+
+.. example::
+
+   The following two lines of code produce the same expression, :manual:`$arrayElemAt </reference/operator/aggregation/arrayElemAt/>`.
+   The ``$`` in front of **categories** tells MongoDB that this is a :manual:`field path </meta/aggregation-quick-reference/#expressions>`,
+   using the **categories** field from the input document.
+
+   .. code-block:: java
+
+      new Document("$arrayElemAt", Arrays.asList("$categories", 0))
+
+   .. code-block:: java
+
+      Document.parse("{ $arrayElemAt: ['$categories', 0] }")
+
+In the following example, the aggregation pipeline uses a
```

Comment:
Would it be helpful to explain all the Projections static functions called or provide links to the api docs for them?

Reply:
I don't think so in this case. I think it will we be more user friendly to ultimately link to the builders pages we create as part of the docs site.